src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/oops/Field.java

```diff
@@ -111,6 +111,11 @@ public static Field[] getFields(InstanceKlass kls) {
     int numJavaFields = crs.readInt();     // read num_java_fields
     int numInjectedFields = crs.readInt(); // read num_injected_fields;
     int numFields = numJavaFields + numInjectedFields;
+    // JumpTable is generated only for classes with > 16 (non-injected) fields
+    if (numJavaFields > 16) {
```

Comment:
The test should use the `JUMP_TABLE_STRIDE` constant.

Reply:
Sure, I can isolate this into static final var, though since this is Java code I can't really take the value from the macro.